### PR TITLE
Fix build issues

### LINKED ISSUE #124: Broken `DriverNotFoundError` calls in `io.py`
### Delft-FIAT version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have checked that this bug exists on the latest version of Delft-FIAT.

### Reproducible Example

https://github.com/Deltares/Delft-FIAT/blob/master/src/fiat/io.py#L835 and https://github.com/Deltares/Delft-FIAT/blob/master/src/fiat/io.py#L1257 call `DriverNotFoundError` with one argument.
`DriverNotFoundError` has 2 arguments

### Current behaviour

Raises an internal exception.

### Desired behaviour

Raise the wanted `DriverNotFoundException`.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by dalmijn**
## Issue addressed
Fixes #124 

## Explanation
Code

## Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `master`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation if needed

## Additional Notes (optional)
No.

### CODE REVIEW

**Comment by dalmijn** [comment]
@Jaapel, Thanks for laying the groundwork. I copied some stuff out of your PR #126 and added it here. I made a separate PR as I really tried to uniformly understand what happened and how to solve it (e.g. with the env variables of gdal).

**Review by Jaapel — APPROVED** [review]

File: .build/hook-fiat.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    datas += list(map(lambda path: (path, "./gdalplugins"), all_plugins))
+
+# Sort out the proj database
+src_proj = None
+if "PROJ_DATA" in os.environ:
+    src_proj = os.environ["PROJ_DATA"]
+elif "PROJ_LIB" in os.environ:
+    src_proj = os.environ["PROJ_LIB"]
+else:
+    logger.warning("Proj data was not found.")
```
Comment (Jaapel) [inline .build/hook-fiat.py:40]: Can you check default locations here?
```suggestion
elif:
  if is_win:
      src_proj = os.path.join(root_path, "Library", "share", "proj")
  else:  # both linux and darwin
      src_proj = os.path.join("share", "proj")
else:
    logger.warning("Proj data was not found.")
```

Don't forget `is_win`: `from PyInstaller.compat import is_win`

File: .build/linux64.sh [hunk 2]
```diff
@@ -1,32 +1,107 @@
...
+  echo $'\t'"-b"$'\t'"Binary name for python environment creation (default: $bin_var)"
+  echo $'\t'"-s"$'\t'"Shell type (default: $shell_var)"
+  echo $'\t'"-h,"$'\t'"Display this help message"$'\n\t'"--help"
+}
+
+# # Check if no arguments are provided
+# if [[ $# -eq 0 ]]; then
+#   help_message
+#   exit 0
+# fi
```
Comment (Jaapel) [inline .build/linux64.sh:26]: remove commented code

File: Dockerfile [hunk 3]
```diff
@@ -12,6 +12,8 @@ RUN curl -fsSL https://pixi.sh/install.sh | bash
 ENV PATH=/home/deltares/.pixi/bin:$PATH
 COPY pixi.toml pyproject.toml README.md ./
 COPY --chown=deltares:deltares src/fiat ./src/fiat
+# Unquote for building inside the docker
+# COPY --chown=deltares:deltares .build ./.build
```
Comment (Jaapel) [inline Dockerfile:16]: `docker run -v .build:.build` >:)

**Review by Jaapel — APPROVED** [review]
Minor comments, I will close mine.

File: Dockerfile (hunk 3 above)
Comment (dalmijn) [reply Dockerfile:16]: Yes.

File: .build/linux64.sh (hunk 2 above)
Comment (dalmijn) [reply .build/linux64.sh:26]: Alright!

File: .build/linux64.sh [hunk 4]
```diff
@@ -1,32 +1,107 @@
...
+# if [ -z "$executable" ]
+# then
+#   # If home_conda is empty, grep with "/home/share"
+#   executable=$(echo "$paths" | grep "^/usr/share")
+# fi
+
+# if [ -z "$conda_executable" ]
+# then
+#   conda_executable="/home/runner/miniconda3/condabin/conda"
+# fi
```
[the remark this answers is not in the document]
Comment (dalmijn) [reply .build/linux64.sh]: Done.

**Review by Jaapel — APPROVED** [review]

**Pull request merged**